src/storage/v2/disk/storage.hpp

```diff
@@ -0,0 +1,253 @@
+
+    void PrepareForNextIndexQuery() override {
+      indexed_vertices_.clear();
+      index_deltas_.clear();
+    }
+
+    std::optional<storage::VertexAccessor> LoadVertexToLabelIndexCache(const rocksdb::Slice &key,
+                                                                       const rocksdb::Slice &value, Delta *index_delta);
+
+    std::optional<storage::VertexAccessor> LoadVertexToMainMemoryCache(const rocksdb::Slice &key,
+                                                                       const rocksdb::Slice &value);
+
+    std::optional<storage::VertexAccessor> LoadVertexToLabelPropertyIndexCache(const rocksdb::Slice &key,
+                                                                               const rocksdb::Slice &value);
+
+    std::optional<storage::EdgeAccessor> DeserializeEdge(const rocksdb::Slice &key, const rocksdb::Slice &value);
+
+   private:
+    VertexAccessor CreateVertex(utils::SkipList<Vertex>::Accessor &accessor, storage::Gid gid,
+                                const std::vector<LabelId> &label_ids, PropertyStore &&properties, Delta *delta);
+
+    void PrefetchEdges(const auto &prefetch_edge_filter);
+
+    Result<EdgeAccessor> CreateEdge(VertexAccessor *from, VertexAccessor *to, EdgeTypeId edge_type, storage::Gid gid,
+                                    std::string_view properties, const std::string &old_disk_key);
+
+    /// Flushes vertices and edges to the disk with the commit timestamp.
+    /// At the time of calling, the commit_timestamp_ must already exist.
+    /// After this method, the vertex and edge caches are cleared.
+    [[nodiscard]] utils::BasicResult<StorageDataManipulationError, void> CheckConstraintsAndFlushMainMemoryCache();
+
+    bool WriteVertexToDisk(const Vertex &vertex);
+    bool WriteEdgeToDisk(EdgeRef edge, const std::string &serializedEdgeKey);
+    bool DeleteVertexFromDisk(const std::string &vertex);
+    bool DeleteEdgeFromDisk(const std::string &edge);
+
+    /// Main storage
+    utils::SkipList<storage::Vertex> vertices_;
+    utils::SkipList<storage::Vertex> indexed_vertices_;
+
+    /// We need them because query context for indexed reading is cleared after the query is done not after the
+    /// transaction is done
+    std::list<Delta> index_deltas_;
+    utils::SkipList<storage::Edge> edges_;
+    Config::Items config_;
+    std::vector<std::string> edges_to_delete_;
+    std::vector<std::pair<std::string, std::string>> vertices_to_delete_;
+    rocksdb::Transaction *disk_transaction_;
+  };
+
+  std::unique_ptr<Storage::Accessor> Access(std::optional<IsolationLevel> override_isolation_level) override {
+    auto isolation_level = override_isolation_level.value_or(isolation_level_);
+    if (isolation_level != IsolationLevel::SNAPSHOT_ISOLATION) {
+      throw utils::NotYetImplemented("Disk storage supports only SNAPSHOT isolation level.");
+    }
+    return std::unique_ptr<DiskAccessor>(new DiskAccessor{this, isolation_level, storage_mode_});
```

Comment:
use make_unique

Reply:
The problem is that make_unique then calls private constructor and compiler says I cannot do that:
In template: calling a private constructor of class 'memgraph::storage::DiskStorage::DiskAccessor'clang(access_ctor)